lib/repl.js

```diff
@@ -361,7 +361,7 @@ function REPLServer(prompt,
 
     // Check to see if a REPL keyword was used. If it returns true,
     // display next prompt and return.
-    if (cmd && cmd.charAt(0) === '.' && isNaN(parseFloat(cmd))) {
+    if (cmd && isNaN(parseFloat(cmd)) && /^\.[a-z]+\s*[^(]*$/.test(cmd)) {
       var matches = cmd.match(/^\.([^\s]+)\s*(.*)$/);
```

Comment:
I think the parse regex is far too lenient to begin with, and that we could probably only use one regex. Furthermore, why don't we just put this within `if (!wasWithinStrLiteral && !isWithinStrLiteral) {` above? That way we can catch it before anything is trimmed.

I'm suggesting something like:

```js
if (!wasWithinStrLiteral && !isWithinStrLiteral) {
  if (cmd && isNaN(parseFloat(cmd)) {
    const matches = /^\.[a-z-_]+\s*[^(){}[];=|&]*$/.exec(cmd);
    ...
  }
  cmd = cmd.trim();
}
```

as an aside, we should probably just check before everything if `(\^\s*$\.test(cmd))` (or `(cmd.trim() === ''`)